src/database/DataTypes/TransactionItem.js

```diff
@@ -72,7 +72,10 @@ export class TransactionItem extends Realm.Object {
    * are finalised, whereas customer invoices immediately influence stock levels.
    * @param {double} quantity The total quantity to set across all batches
    */
-  setTotalQuantity(database, quantity) {
+   // TODO: document function param
+  setTotalQuantity(database, requestedQuantity, referenceQuantity = -1) {
+    let quantity = requestedQuantity;
+    if (referenceQuantity > 0) quantity = Math.min(requestedQuantity, referenceQuantity);
```

Comment:
As per Edwin's comment, `My belief is that in a case where you should always expect a condition to be met, you shouldn't safety check for that condition. If you do, there may be something going wrong that is silenced by your safety check. If you don't, the underlying problem is very brutally and obviously exposed by a crash (and a bugsnag report! heh)` in https://github.com/sussol/mobile/pull/450, I don't see a case where setTotalQuanity is called on finalised Transaction. If we were to check for finalised here, we might as well check it for every data type function... and that's a bit of refactoring